src/js/notes.md

```diff
@@ -0,0 +1,179 @@
+# Gleam to JS
+
+Working through the Gleam book to see how the language concepts translate to JS.
+
+### Comments, Strings, Bools, Ints and Floats
+These should all be trivial to map.
+
+*Integers in JS are not "efficient" but all the Gleam operators should work appropriately and return more integers.*
+
+From Gleam book and translated JS
+```rs
+1 + 1 // => 2
+5 - 1 // => 4
+5 / 2 // => 2
+3 * 3 // => 9
+5 % 2 // => 1
+```
+
+```js
+1 + 1 // => 2
+5 - 1 // => 4
+parseInt(5 / 2) // => 2
+3 * 3 // => 9
+5 % 2 // => 1
+```
+
+JS has no way to check if something is a float or integer reliably.
+`Number.isInteger(1.0) // => false`
+
+### Let bindings
+JS same.
+
+### Lists
+
+*JS lists are Arrays, so they will not have the same performance charachteristics.*
+However they match very neatly for syntax
```

Comment:
I think we may need to look into implementing immutable lists in some fashion, otherwise all our recursive algorithms will become very slow.

I'm interested to see what Bucklescript and Purecript do here.

Reply:
I think that's probably a good idea. Although for version one I am very happy with slow but working